### ISSUE TITLE
Detect overriden methods in templated classes and suppress fix for UnnecessaryValueParam check

### ISSUE THREAD

**Issue by fberger** [llvm/llvm-project#30584]
|  |  |
| --- | --- |
| Bugzilla Link | [31236](https://llvm.org/bz31236) |
| Version | unspecified |
| OS | Linux |
| CC | @pepsiman |

**Comment by pepsiman** [comment]
e.g.

```c++
template <typename T>
struct NegativeDependentTypeInterface {
  virtual void Method(ExpensiveToCopyType E) = 0;
};

template <typename T>
struct NegativeOverrideImpl : public NegativeDependentTypeInterface<T> {
  void Method(ExpensiveToCopyType E) {}
};
```

shouldn't trigger performance-unnecessary-value-param warnings for the overriding method.

**Comment by PiotrZSL** [comment]
Related to 55665

**Comment by vbvictor** [comment]
The check already respect override: https://godbolt.org/z/bMj8TWaca.
Approximation is an addition
